plugins/cpu/cpu.go

```diff
@@ -13,31 +21,170 @@ const (
 	niceGauge
 )
 
+// Hold cpu sampled data and calculate moving averages
+type sample struct {
+	alpha        float64
+	values       []float64
+	expMovingAvg float64
+}
+
+func newSample(alpha float64) *sample {
+	return &sample{alpha: alpha}
+}
+
+func (s *sample) add(v float64) {
+	s.values = append(s.values, v)
+
+	if s.expMovingAvg == 0 {
+		s.expMovingAvg = v
+	} else {
+		s.expMovingAvg = v*s.alpha + (s.expMovingAvg * (1 - s.alpha))
+	}
+}
+
+// Reset data at reporting intervals
+func (s *sample) drain() float64 {
+	avg := s.expMovingAvg
+	s.expMovingAvg = 0
```

Comment:
I was resetting it to get the average utilization over the reporting period. I tried it out with both reset and long lived and data is below cpu sampled every second and reporting every 10s.
```
ts, average (reset), average (long lived)
2016/03/15 16:38:11, 7.096684, 7.096684 
2016/03/15 16:38:21, 6.835846, 8.185538 
2016/03/15 16:38:31, 15.642729, 17.186029 
2016/03/15 16:38:41, 30.054096, 26.736691 
2016/03/15 16:38:51, 30.031195, 29.899487 
2016/03/15 16:39:01, 30.095461, 30.971534 
2016/03/15 16:39:11, 16.841192, 18.377162 
2016/03/15 16:39:21, 5.724024, 11.345251 
2016/03/15 16:39:31, 6.059904, 8.714710 
2016/03/15 16:39:41, 6.568810, 7.680073 
2016/03/15 16:39:51, 5.381379, 7.010598 
2016/03/15 16:40:01, 6.045813, 7.446830 
2016/03/15 16:40:11, 5.845820, 7.741508 
```